src/App.tsx

```diff
@@ -1,223 +1,239 @@
+  const [isAllSelected, setIsAllSelected] = useState(false);
+
+  // Geting ↓
+
+  const getTodosFromServer = async () => {
+    setIsLoading(true);
+    try {
+      if (user?.id) {
+        const todosFromServer = await getTodos(user?.id);
+
+        setVisibleTodos(todosFromServer);
+        setTodos(todosFromServer);
+      }
+    } catch {
+      setErrorMessage('upload todos');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  // Deleting ↓
+
+  const deleteInLocalTodos = (id: number) => {
+    const filteredTodos = todos.filter(todo => todo.id !== id);
+
+    setVisibleTodos(filteredTodos);
+    setTodos(filteredTodos);
+  };
+
+  const onDeleteTodo = async (id: number) => {
+    setloadingTodoId(id);
+    try {
+      await deleteTodo(id);
+      deleteInLocalTodos(id);
+    } catch {
+      setErrorMessage('delete a todo');
+    } finally {
+      setloadingTodoId(null);
+    }
+  };
+
+  const clearCompleted = () => {
+    const filteredTodos = todos.filter((todo) => {
+      const { completed, id } = todo;
+
+      if (completed) {
+        deleteTodo(id);
+      }
+
+      return !completed;
+    });
+
+    setTodos(filteredTodos);
+    setVisibleTodos(filteredTodos);
+  };
+
+  // Adding ↓
+
+  const addToLocalTodos = (newTodo: Todo) => {
+    setVisibleTodos([...todos, newTodo]);
+    setTodos([...todos, newTodo]);
+  };
+
+  const addTodo = async (title: string) => {
+    setIsAdding(true);
+    setloadingTodoId(0);
+    try {
+      const newTodo: Todo = await createTodo(user?.id, title.trim());
+
+      addToLocalTodos(newTodo);
+    } catch {
+      setErrorMessage('add a todo');
+    } finally {
+      setIsAdding(false);
+      setloadingTodoId(null);
+    }
+  };
+
+  // Updating ↓
+
+  const toggleStatus = (todoId: number, comleted: boolean) => {
+    const index = visibleTodos.findIndex(todo => todo.id === todoId);
+
+    setVisibleTodos((prevVisibleTodos) => {
+      const todosCopy = [...prevVisibleTodos];
+
+      todosCopy[index].completed = !comleted;
+
+      return todosCopy;
+    });
+  };
+
+  const selectAllTodos = () => {
+    const selectedTodos = visibleTodos.map((todo) => {
+      const { completed, id } = todo;
+
+      if (isAllSelected) {
+        return { ...todo, completed: false };
+      }
+
+      if (completed) {
+        return todo;
+      }
+
+      updatingTodoCompleted(id, true);
+
+      return { ...todo, completed: true };
+    });
+
+    setTodos(selectedTodos);
+    setVisibleTodos(selectedTodos);
+  };
+
+  const changeTitle = (todoId: number, newTitle: string) => {
+    const index = visibleTodos.findIndex(todo => todo.id === todoId);
+
+    setVisibleTodos((prevVisibleTodos) => {
+      const todosCopy = [...prevVisibleTodos];
+
+      todosCopy[index].title = newTitle;
+
+      return todosCopy;
+    });
+  };
+
+  // Hooks ↓
+
+  useEffect(() => {
+    const countofLeftItems = visibleTodos
+      .filter(({ completed }) => !completed).length;
```

Comment:
here you filtring different completed status. Not only those which `active` but those who had another `completed` status, and as a result you have different numbers
![image](https://user-images.githubusercontent.com/107215656/194337120-e02c5cb2-0fb0-40cf-a511-d0eba933cf18.png)
![image](https://user-images.githubusercontent.com/107215656/194337183-ba7fb98b-0f3c-4912-be22-ceff3adb343c.png)